core/store/src/trie/shard_tries.rs

```diff
@@ -60,22 +63,36 @@ impl StateSnapshot {
         prev_block_hash: CryptoHash,
         flat_storage_manager: FlatStorageManager,
         shard_uids: &[ShardUId],
+        block: Option<&Block>,
```

Comment:
note: A bit unfortunate that we need the whole block just for chunks metadata, but I don't see a better option.